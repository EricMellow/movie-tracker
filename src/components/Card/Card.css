.movieCard {
  display: flex;
  align-items: center;
  width: 94%;
  height: 90px;
  margin-bottom: 10px;
  padding: 10px;
  position: relative;
}

.selected, .movieCard.selected:hover {
  padding: 10px;
  background: #111;
  border-radius: 5px;
  box-shadow: inset 1px 1px 1px #000;
}


.movieCard:hover {
  cursor: pointer;
  background: #1d1d1d;
  border-radius: 5px;
}

.movieCard img {
  height: 100%;
  width: auto;
  margin-right: 20px;
  border-radius: 5px;
}

.movieCard p {
  font-size: 0.8rem;
  color: #9f9f9f
}

h3 {
  font-size: 0.8rem;
  text-transform: uppercase;
  width: 75%;
}

.stars {
  border: 0;
  float: left;
  position: relative;
  width: 76px;
} 

.goldStars{
  border: 0;
  position: absolute;
  top: 0;
  left: 0;
  overflow: hidden;
 } 

 .stars img {
   max-width: 76px;
   height: auto;
 }

 .cardContainer {
   width: 100%;
 }

 hr {
  margin-top: 0px;
  background: #333;
  height: 1px;
  border: none;
}

.selected .triangle {
  background-color: #3c4b65;
}

.triangle {
	background-color: #333;
  text-align: left;
  height: 21px;
  right: 15px;
  position: absolute;
}
.triangle:before,
.triangle:after {
	content: '';
	position: absolute;
	background-color: inherit;
}
.triangle,
.triangle:before,
.triangle:after {
	width:  0.7em;
	height: 0.7em;
	border-top-right-radius: 30%;
}

.triangle {
	transform: rotate(30deg) skewX(-30deg) scale(1,.866);
}
.triangle:before {
	transform: rotate(-135deg) skewX(-45deg) scale(1.414,.707) translate(0,-50%);
}
.triangle:after {
	transform: rotate(135deg) skewY(-45deg) scale(.707,1.414) translate(50%);
}

@media only screen and (max-width: 1200px) {
  .movieCard {
    flex-direction: column;
    height: 155px;
  }

  img.thumbImg {
    height: 57%;
    margin: 5px 0 0 0;
  }

  .triangle {
    display: none;
  }

  .movieCard h3 {
    width: 100%;
  }

  .movieCard div {
    text-align: center;
  }
}