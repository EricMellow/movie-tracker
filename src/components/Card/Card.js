import React, { Component } from "react";
import './Card.css';
import { connect } from 'react-redux';
import { setSelectedMovieId } from '../../actions/index';
import PropTypes from 'prop-types';

export class Card extends Component {
  constructor(props) {
    super(props);
  }

  clickHandler = () => {
    const id = this.props.id;
    this.props.storeSelectedMovieId(id);
  }

  render() {
    const rating = this.props.rating * 10;
    const cardClass = this.props.selectedMovieId === this.props.id ? "movieCard selected" : "movieCard";
    const img = `https://image.tmdb.org/t/p/w500/${this.props.backdrop}`;
    
    return (
      <div
        className={cardClass}
        onClick={this.clickHandler}
      >
        <img src={img} className="thumbImg"/>
        <div className="titleStars">
          <h3>{this.props.title}</h3>
          <div className="stars">
            <div>
              <img src={require('../Card/stars-gray.png')} />
            </div>
            <div className="goldStars" style={{"width" : `${rating}%`}}>
              <img src={require('../Card/stars-gold.png')}  />
            </div>
          </div>
        </div>
        <div className='triangle'></div>
      </div>
    );
  }
};

Card.propTypes = {
  storeSelectedMovieId: PropTypes.func,
  title: PropTypes.string,
  id: PropTypes.number,
  rating: PropTypes.number,
  selectedMovieId: PropTypes.number,
  backdrop: PropTypes.string
};

export const mapStateToProps = (state)=>({
  selectedMovieId: state.selectedMovieId
});

export const mapDispatchToProps = (dispatch) => ({
  storeSelectedMovieId: (id) => dispatch(setSelectedMovieId(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);