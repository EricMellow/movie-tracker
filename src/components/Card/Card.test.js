import React from 'react';
import { Card, mapStateToProps, mapDispatchToProps } from "./Card";
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockStoreSelectedMovieId;

  beforeEach(() => {
    mockStoreSelectedMovieId = jest.fn();

    wrapper = shallow(<Card storeSelectedMovieId={mockStoreSelectedMovieId} /> );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('clickHandler', () => {
    it('should call storeSelectedMovieId', () => {
      wrapper.instance().clickHandler();

      expect(wrapper.instance().props.storeSelectedMovieId).toHaveBeenCalled();
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params on storeSelectedMovieId', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'SET_SELECTED_MOVIE_ID',
        movieId: 182935
      };
      
      mappedProps.storeSelectedMovieId(182935);
  
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });

  describe('mapStateToProps', () => {
    it('should map selectedMovieId to props', () => {
      const state = {
        selectedMovieId: 12345
      };
      const expected = {
        selectedMovieId: 12345
      };
      const mappedProps = mapStateToProps(state);
      
      expect(mappedProps).toEqual(expected);
    });
  });
});