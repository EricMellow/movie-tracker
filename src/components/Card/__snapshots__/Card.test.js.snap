// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    storeSelectedMovieId={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="thumbImg"
          src="https://image.tmdb.org/t/p/w500/undefined"
/>,
        <div
          className="titleStars"
>
          <h3 />
          <div
                    className="stars"
          >
                    <div>
                              <img
                                        src="stars-gray.png"
                              />
                    </div>
                    <div
                              className="goldStars"
                              style={
                                        Object {
                                                  "width": "NaN%",
                                                }
                              }
                    >
                              <img
                                        src="stars-gold.png"
                              />
                    </div>
          </div>
</div>,
        <div
          className="triangle"
/>,
      ],
      "className": "movieCard selected",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "thumbImg",
          "src": "https://image.tmdb.org/t/p/w500/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <div
              className="stars"
>
              <div>
                            <img
                                          src="stars-gray.png"
                            />
              </div>
              <div
                            className="goldStars"
                            style={
                                          Object {
                                                        "width": "NaN%",
                                                      }
                            }
              >
                            <img
                                          src="stars-gold.png"
                            />
              </div>
</div>,
          ],
          "className": "titleStars",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <img
                                    src="stars-gray.png"
                  />
</div>,
                <div
                  className="goldStars"
                  style={
                                    Object {
                                                      "width": "NaN%",
                                                    }
                  }
>
                  <img
                                    src="stars-gold.png"
                  />
</div>,
              ],
              "className": "stars",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="stars-gray.png"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "stars-gray.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="stars-gold.png"
/>,
                  "className": "goldStars",
                  "style": Object {
                    "width": "NaN%",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "stars-gold.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "triangle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="thumbImg"
            src="https://image.tmdb.org/t/p/w500/undefined"
/>,
          <div
            className="titleStars"
>
            <h3 />
            <div
                        className="stars"
            >
                        <div>
                                    <img
                                                src="stars-gray.png"
                                    />
                        </div>
                        <div
                                    className="goldStars"
                                    style={
                                                Object {
                                                            "width": "NaN%",
                                                          }
                                    }
                        >
                                    <img
                                                src="stars-gold.png"
                                    />
                        </div>
            </div>
</div>,
          <div
            className="triangle"
/>,
        ],
        "className": "movieCard selected",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "thumbImg",
            "src": "https://image.tmdb.org/t/p/w500/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <div
                className="stars"
>
                <div>
                                <img
                                                src="stars-gray.png"
                                />
                </div>
                <div
                                className="goldStars"
                                style={
                                                Object {
                                                                "width": "NaN%",
                                                              }
                                }
                >
                                <img
                                                src="stars-gold.png"
                                />
                </div>
</div>,
            ],
            "className": "titleStars",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <img
                                        src="stars-gray.png"
                    />
</div>,
                  <div
                    className="goldStars"
                    style={
                                        Object {
                                                            "width": "NaN%",
                                                          }
                    }
>
                    <img
                                        src="stars-gold.png"
                    />
</div>,
                ],
                "className": "stars",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      src="stars-gray.png"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "src": "stars-gray.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      src="stars-gold.png"
/>,
                    "className": "goldStars",
                    "style": Object {
                      "width": "NaN%",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "src": "stars-gold.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "triangle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
